chamge/amqp-arbiter-backend.c

```diff
@@ -305,6 +356,214 @@ chamge_amqp_arbiter_backend_deactivate (ChamgeArbiterBackend * arbiter_backend)
+          amqp_error_string2 (r));
+      goto out;
+    }
+    g_debug ("published to queue[%s], exchange[%s], request[%s]", queue_name,
+        exchange, request);
+  }
+
+  /* wait an answer */
+  if (amqp_basic_consume (amqp_conn, 1, amqp_reply_queue, amqp_empty_bytes, 0,
+          1, 0, amqp_empty_table) == NULL) {
+    g_set_error (error, CHAMGE_BACKEND_ERROR,
+        CHAMGE_BACKEND_ERROR_OPERATION_FAILURE, "basic consume failure >> %s",
+        _amqp_get_rpc_reply_string (amqp_get_rpc_reply (amqp_conn)));
+    goto out;
+  }
+
+
+  /* wait an answer */
+  do {
+    amqp_rpc_reply_t amqp_r;
+    amqp_envelope_t envelope;
+    struct timeval timeout = { 10, 0 };
+
+    amqp_maybe_release_buffers (amqp_conn);
+
+    g_debug ("Wait for response message from [%s]",
+        (gchar *) amqp_reply_queue.bytes);
+    amqp_r = amqp_consume_message (amqp_conn, &envelope, &timeout, 0);
+
+    if (amqp_r.reply_type == AMQP_RESPONSE_LIBRARY_EXCEPTION
+        && amqp_r.library_error == AMQP_STATUS_TIMEOUT) {
+      g_set_error_literal (error, CHAMGE_BACKEND_ERROR,
+          CHAMGE_BACKEND_ERROR_OPERATION_FAILURE, "consume msg timeout");
+      break;
+    }
+
+    if (amqp_r.reply_type != AMQP_RESPONSE_NORMAL) {
+      g_set_error (error, CHAMGE_BACKEND_ERROR,
+          CHAMGE_BACKEND_ERROR_OPERATION_FAILURE, "consume msg failure >> %s",
+          _amqp_get_rpc_reply_string (amqp_r));
+      break;
+    }
+
+    g_debug ("Delivery %u, exchange %.*s routingkey %.*s",
+        (unsigned) envelope.delivery_tag, (int) envelope.exchange.len,
+        (char *) envelope.exchange.bytes, (int) envelope.routing_key.len,
+        (char *) envelope.routing_key.bytes);
+
+    if ((envelope.message.properties._flags & AMQP_BASIC_CORRELATION_ID_FLAG) ==
+        0) {
+      g_debug ("discard >> correaltion id is not exist");
+      continue;
+    }
+
+    if (strncmp (correlation_id,
+            envelope.message.properties.correlation_id.bytes,
+            envelope.message.properties.correlation_id.len) != 0) {
+
+      g_debug ("discard >> correaltion id is not matched [%s] [%.*s]",
+          correlation_id, (gint) envelope.message.properties.correlation_id.len,
+          (gchar *) envelope.message.properties.correlation_id.bytes);
+      continue;
+    }
+
+    g_debug ("Correlation ID : %.*s",
+        (int) envelope.message.properties.correlation_id.len,
+        (char *) envelope.message.properties.correlation_id.bytes);
+    if (envelope.message.properties._flags & AMQP_BASIC_CONTENT_TYPE_FLAG) {
+      g_debug ("Content-type: %.*s",
+          (int) envelope.message.properties.content_type.len,
+          (char *) envelope.message.properties.content_type.bytes);
+    }
+
+    if (*response != NULL) {
+      g_free (*response);
+      *response = NULL;
+    }
+    *response =
+        g_strndup (envelope.message.body.bytes, envelope.message.body.len);
+
+    amqp_destroy_envelope (&envelope);
+    break;
+  } while (1);
```

Comment:
We shouldn't use an infinite loop in a function. Instead, `g_idle_add()` is recommended.
but no change is required for this time. We can fix it later.